parsers/MD.py

```diff
@@ -147,21 +146,17 @@ def get_archive_data(session: Session | None = None, dates=None) -> list:
 
     archive_url = archive_base_url
     if date1 and date2:
```

Comment:
Well this is weird.

The function expects 2 dates in a tuple or list but it's ever only called with 1...

Not only should this never happen IMO but it's just causing the one date to be passed to be duplicated...

However the date passed itself is a datetime object, not a string which is causing `.fromisoformat()` to fail.

You should be able to change it to just accept one date in the function and then call fromisoformat directly on it.

Reply:
Oh wow... thanks for catching that! The function calls in `__main__` feeding in strings threw me off!

Should hopefully all be sorted now